staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go

```diff
@@ -122,22 +128,153 @@ var (
 )
 
 type claimsTest struct {
-	name        string
-	options     Options
-	now         time.Time
-	signingKey  *jose.JSONWebKey
-	pubKeys     []*jose.JSONWebKey
-	claims      string
-	want        *user.DefaultInfo
-	wantSkip    bool
-	wantErr     bool
-	wantInitErr bool
+	name               string
+	options            Options
+	now                time.Time
+	signingKey         *jose.JSONWebKey
+	pubKeys            []*jose.JSONWebKey
+	claims             string
+	want               *user.DefaultInfo
+	wantSkip           bool
+	wantErr            bool
+	wantInitErr        bool
+	claimToResponseMap map[string]string
+	openIDConfig       string
+}
+
+// Replace formats the contents of v into the provided template.
+func replace(tmpl string, v interface{}) string {
+	t := template.Must(template.New("test").Parse(tmpl))
+	buf := bytes.NewBuffer(nil)
+	t.Execute(buf, &v)
+	ret := buf.String()
+	glog.V(4).Infof("Replaced: %v into: %v", tmpl, ret)
+	return ret
+}
+
+// newClaimServer returns a new test HTTPS server, which is rigged to return
+// OIDC responses to requests that resolve distributed claims. signer is the
+// signer used for the served JWT tokens.  claimToResponseMap is a map of
+// responses that the server will return for each claim it is given.
+func newClaimServer(t *testing.T, keys jose.JSONWebKeySet, signer jose.Signer, claimToResponseMap map[string]string, openIDConfig *string) *httptest.Server {
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		glog.V(5).Infof("request: %+v", *r)
+		switch r.URL.Path {
+		case "/.testing/keys":
+			w.Header().Set("Content-Type", "application/json")
+			keyBytes, err := json.Marshal(keys)
+			if err != nil {
+				t.Fatalf("unexpected error while marshaling keys: %v", err)
+			}
+			glog.V(5).Infof("%v: returning: %+v", r.URL, string(keyBytes))
+			w.Write(keyBytes)
+
+		case "/.well-known/openid-configuration":
+			w.Header().Set("Content-Type", "application/json")
+			glog.V(5).Infof("%v: returning: %+v", r.URL, *openIDConfig)
+			w.Write([]byte(*openIDConfig))
+		// These claims are tested in the unit tests.
+		case "/groups":
+			fallthrough
+		case "/rabbits":
+			if claimToResponseMap == nil {
+				glog.Errorf("no claims specified in response")
+				t.FailNow()
```

Comment:
Sorry, I linked to FailNow because of it's godoc. Not because fatal should be replaced with it.

> Fatal is equivalent to Log followed by FailNow.

and

> ... FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

Need to replace this with Errorf calls